### ISSUE TITLE
Document `tool_requires(..., options={})` and other methods of setting build requirement options

### ISSUE THREAD

**Issue by samuel-emrys** [conan-io/docs#3026]
The 2.0 documentation doesn't address any methods of setting the options of build requirements, either `tool_requires` or `test_requires`. The 1.x method of setting them in `configure()` doesn't appear to be working in 2.0 (not sure if this is a bug), but the documentation should address at least some methods of doing this.

The following links refer to the `tool_requires(..., options={})` syntax which is currently undocumented as far as I can tell:
* https://github.com/conan-io/conan/issues/3213#issuecomment-1440979025
* https://github.com/conan-io/conan/issues/13215#issuecomment-1442617425

I had a feeling that I'd seen another method of setting options for build requirements, but I can't find the issue at the moment. It strikes me that there _must_ be one, because as it stands the above syntax is not sufficient to modify package options based on logic.

For completeness, this appears to work:

```python
def build_requirements(self):
    self.tool_requires("mytool@user/channel",options={"optA":"value"})
```

**Comment by uyha** [comment]
`tool_requires` and `test_requires` are documented [here](https://docs.conan.io/2/reference/conanfile/methods/build_requirements.html) but they all refer to `requires` which is undocumented for now AFAIK.

**Comment by samuel-emrys** [comment]
@uyha yes, thanks - I found those in the documentation. This ticket is not about documenting `tool_requires` generally, but about documenting ways of setting options on `tool_requires` packages from consuming recipes.

**Comment by roalter** [comment]
Would be great if somehow this would still be possible in 1.x. 
Still strugling on updating to 2.x.
